### PR TITLE
Parameter.Expand consumes optional StepsVariables argument

### PR DESCRIPTION
**Pull request by rihter007**
Minimal change to make step variables feature fully work in prod

### CODE REVIEW

File: pkg/test/functions.go [hunk 1]
```diff
@@ -32,6 +32,23 @@ func getFuncMap() map[string]interface{} {
 	return mapCopy
 }
 
+func registerStepVariableAccessor(fm map[string]interface{}, tgtID string, vars StepsVariablesReader) {
```
Comment (mimir-d) [inline pkg/test/functions.go:35]: dont these always expand to a string? the descriptor json is a string, what's the point of having an int version? just type safety?

File: pkg/test/parameter_test.go [hunk 2]
```diff
@@ -46,10 +48,60 @@ func TestParameterExpandUserFunctions(t *testing.T) {
...
+		res, err := p.Expand(&target.Target{FQDN: x[1], ID: x[2]}, nil)
 		require.NoError(t, err, x[0])
 		require.Equal(t, x[3], res, x[0])
 	}
 	require.NoError(t, UnregisterFunction("CustomFunc"))
 	require.Error(t, UnregisterFunction("NoSuchFunction"))
 }
+
+func TestStepVariablesExpand(t *testing.T) {
+	p := NewParam("{{ StringVar \"step1\" \"string_var\" }}: {{ IntVar \"step1\" \"int_var\" }}")
```
Comment (mimir-d) [inline pkg/test/parameter_test.go:60]: i thought the naming we agreed to was with dot separated ids.
```suggestion
	p := NewParam("{{ StringVar \"step1.string_var\" }}: {{ IntVar \"step1.int_var\" }}")
```

i see in the Get() below that they are joined with dot

**Review by mimir-d — CHANGES_REQUESTED** [review]

File: pkg/test/parameter_test.go (hunk 2 above)
Comment (rihter007) [reply pkg/test/parameter_test.go:60]: Yes, originally we agreed on dots. I just later thought to give it a try as two parameters.
Probably with dot is better.
Will rework!

File: pkg/test/functions.go (hunk 1 above)
Comment (rihter007) [reply pkg/test/functions.go:35]: yes, always to a string or string and error: https://pkg.go.dev/text/template#FuncMap

**Review by mimir-d — APPROVED** [review]

**Pull request merged**